src/introspection/src5.cairo

```diff
@@ -14,37 +14,44 @@ mod SRC5 {
         const INVALID_ID: felt252 = 'SRC5: invalid id';
     }
 
-    #[external(v0)]
-    impl SRC5Impl of interface::ISRC5<ContractState> {
-        fn supports_interface(self: @ContractState, interface_id: felt252) -> bool {
+    #[embeddable_as(SRC5Impl)]
+    impl SRC5<
+        TContractState, +HasComponent<TContractState>
+    > of interface::ISRC5<ComponentState<TContractState>> {
+        fn supports_interface(
+            self: @ComponentState<TContractState>, interface_id: felt252
+        ) -> bool {
             if interface_id == interface::ISRC5_ID {
                 return true;
             }
             self.SRC5_supported_interfaces.read(interface_id)
         }
     }
 
-    #[external(v0)]
-    impl SRC5CamelImpl of interface::ISRC5Camel<ContractState> {
-        fn supportsInterface(self: @ContractState, interfaceId: felt252) -> bool {
-            SRC5Impl::supports_interface(self, interfaceId)
+    #[embeddable_as(SRC5CamelImpl)]
+    impl SRC5Camel<
+        TContractState, +HasComponent<TContractState>
+    > of interface::ISRC5Camel<ComponentState<TContractState>> {
+        fn supportsInterface(self: @ComponentState<TContractState>, interfaceId: felt252) -> bool {
+            self.supports_interface(interfaceId)
```

Comment:
I think now is a good time, yes.

Reply:
Actually, I started the change, and is not small, because it touches a couple of modules, and I think is better to have it on a different PR anyway, to include it in the changelog, wdyt?

Reply:
Sounds good to me!